enhancements/evidence-forwarding.md

```diff
@@ -0,0 +1,305 @@
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could configure both storage mechanisms for redundancy:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage-and-cluster-storage
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+### User Stories
+
+1. As a site reliability engineer, I want to configure the Compliance Operator
+   to send all ARF reports to a dedicated object store.
+2. As a site reliability engineer, I want to opt out of storing any ARF reports
+   on physical volumes.
+3. As a site reliability engineer, I expect verbose alerts when the Compliance
+   Operator cannot send ARF reports to a configured external storage pool.
+4. As a site reliability engineer, I expect the Compliance Operator to fail if
+   the configured endpoint is unreachable.
+5. As a site reliability engineer, I expect to forward evidence to
+   public object-storage endpoints, like AWS S3.
+5. As a site reliability engineer, I expect to forward evidence to
+   private object-storage endpoints in cluster, Ceph or OpenShift Data Foundation.
+
+### API Extensions
+
+This proposal would extend the `ScanSetting` custom resource definition (CRD)
+to include an additional attribute called `externalResultStorage`.
+
+## `externalResultStorage`
+
+This attribute is an *optional* argument to `ScanSetting` CRDs. Existing
+instances of `ScanSetting` custom resources will be treated as if they're not
+configured to use external storage, making the feature backwards compatible.
+
+###  `endpoint`
+
+The `endpoint` is a required argument for `externalResultStorage`. The
+Compliance Operator will validate the `endpoint` at runtime (e.g., when a
+`ScanSetting` is created or used) to ensure connectivity. This doesn't
+necessarily mean the endpoint will always be available, but it is important to
+fail early if the URL is malformed.
+
+###  `tls`
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+There is a chance that we will have requests for different forwarding
+implementations. We should abstract the forwarding interface from the actual
+forwarding implementation, which can be loaded when the operator starts.
+
+### Risks and Mitigations
+
+One potential risk users face when using this feature is the availability of
+the storage endpoint. What should the Compliance Operator do if it needs to
+forward evidence, but the endpoint isn't available?
+
+The operator should absolutely emit an alert.
+
+The configuration described above can include storing the evidence on a
```

Comment:
> The configuration described above can include storing the evidence on a physical volume as well as externally.

I read this as meaning we would be able to configure the options above to point to the resultserver component to store to a PV.. ?

I think that the configuration addition should only be valid for an external store. We've got physical volume storage by default resultserver/pv deal. And we'll add an option to turn that off, but I think we intend to keep them separate.